src/dpdk_trex_port_attr.cpp

```diff
@@ -222,6 +222,27 @@ int DpdkTRexPortAttr::set_multicast(bool enable){
     return 0;
 }
 
+int DpdkTRexPortAttr::set_hardware_timesync(bool enable){
+    int ret;
+    if (enable) {
+        ret = rte_eth_timesync_enable(m_repid);
+        if (ret < 0) {
+            hardware_timesync_enabled = false;
+            printf("Hardware timesync enable failed: %d\n", ret);
+        } else {
+            hardware_timesync_enabled = true;
+        }
+    } else {
+        ret = rte_eth_timesync_disable(m_repid);
+        if (ret < 0) {
+            printf("Hardware timesync disable failed: %d\n", ret);
+        } else {
+            hardware_timesync_enabled = false;
+        }
+    }
+    return ret;
+}
```

Comment:
I think with enabling timesync we need to set up initial time to it. (?)

Reply:
hmm I don't see anything like this in ptpclient example https://git.dpdk.org/dpdk/tree/examples/ptpclient/ptpclient.c#n243. And we can't check how it really works until we have servers with hardware timestamping support...

Reply:
Ok, fair enough